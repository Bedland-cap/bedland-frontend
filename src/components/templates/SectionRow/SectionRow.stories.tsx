import type { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import SectionRow from './SectionRow';

const meta = {
  title: 'templates/SectionRow',
  component: SectionRow,
  tags: ['autodocs'],
  decorators: [(Story) => <BrowserRouter>{Story()}</BrowserRouter>],
} satisfies Meta<typeof SectionRow>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: (
      <>
        <div
          style={{
            border: '1px solid white',
            height: '300px',
            width: '300px',
            color: 'white',
          }}
        >
          this is
        </div>
        <div
          style={{
            border: '1px solid white',
            height: '300px',
            width: '300px',
            color: 'white',
          }}
        >
          template
        </div>
      </>
    ),
  },
};
