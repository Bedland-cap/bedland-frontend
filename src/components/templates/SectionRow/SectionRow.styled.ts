import styled from 'styled-components';

const SectionRowLayout = styled.div`
  align-items: center;
  display: flex;
  gap: 2rem;
  height: 100%;
  justify-content: space-around;
  max-width: 1600px;
  padding: 0 2rem;
`;
export default SectionRowLayout;
