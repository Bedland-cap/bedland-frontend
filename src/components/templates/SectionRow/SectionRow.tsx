import { PropsWithChildren } from 'react';
import SectionRowLayout from './SectionRow.styled';

const SectionRow = ({ children }: PropsWithChildren) => (
  <SectionRowLayout>{children}</SectionRowLayout>
);

export default SectionRow;
