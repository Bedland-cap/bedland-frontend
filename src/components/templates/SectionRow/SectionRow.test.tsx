import renderer from 'react-test-renderer';
import SectionRow from './SectionRow';

describe('Login Page Template', () => {
  it('renders correctly', () => {
    const snapshot = renderer
      .create(
        <SectionRow>
          <div>renders correctly</div>
        </SectionRow>,
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
