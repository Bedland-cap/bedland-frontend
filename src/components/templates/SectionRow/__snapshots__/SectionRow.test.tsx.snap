// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page Template renders correctly 1`] = `
<div
  className="sc-bcXHqe cipXCG"
>
  <div>
    renders correctly
  </div>
</div>
`;
